modApiServer/src/org/aion/api/server/ApiAion.java

```diff
@@ -56,18 +47,34 @@
 import org.aion.zero.types.AionTransaction;
 import org.aion.zero.types.AionTxReceipt;
 
-public abstract class ApiAion extends Api {
+import java.math.BigInteger;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
-    protected IAionChain ac = null;
+public abstract class ApiAion extends Api {
+    protected NrgOracle nrgOracle;
+    protected IAionChain ac;
     protected final static short FLTRS_MAX = 1024;
-    protected AtomicLong fltrIndex = new AtomicLong(1);
+    protected final long DEFAULT_NRG_LIMIT = 500_000L;
+    protected AtomicLong fltrIndex = new AtomicLong(0);
     protected Map<Long, Fltr> installedFilters = null;
     protected Map<ByteArrayWrapper, AionTxReceipt> pendingReceipts;
+    protected String[] compilers = new String[] {"solidity"};
 
     public ApiAion(final IAionChain _ac) {
         this.ac = _ac;
         this.installedFilters = new ConcurrentHashMap<>();
-        this.regEvents();
+
+        // register events
+        IEventMgr evtMgr = this.ac.getAionHub().getEventMgr();
+        evtMgr.registerEvent(Collections.singletonList(new EventTx(EventTx.CALLBACK.PENDINGTXUPDATE0)));
+        evtMgr.registerEvent(Collections.singletonList(new EventBlock(EventBlock.CALLBACK.ONBLOCK0)));
```

Comment:
`ONBLOCK0` or `ONBEST0`?

Reply:
Naming is confusing. CALLBACK.ONBLOCK0 triggers event every block. Correct behaviour.